pandas/core/arrays/datetimes.py

```diff
@@ -376,6 +377,16 @@ def _resolution(self):
     # ----------------------------------------------------------------
     # Array-like Methods
 
+    def __array__(self, dtype=None):
+        if is_object_dtype(dtype):
+            return np.array(list(self), dtype=object)
+        elif is_int64_dtype(dtype):
+            return self.asi8
+
+        # TODO: warn that dtype is not used?
+        #  warn that conversion may be lossy?
+        return self._data.view(np.ndarray)  # follow Index.__array__
```

Comment:
Other question: what is the `.view(np.ndarray)` part doing if it is already an array? Can we remove it?

Reply:
It can probably be removed; this is taken directly from the `Index.__array__` implementation, so I think the maybe-removing this should be done at the same time those methods are overhauled (ill be opening an Issue shortly)